splinterd/src/main.rs

```diff
@@ -97,65 +102,116 @@ fn create_config(_toml_path: Option<&str>, _matches: ArgMatches) -> Result<Confi
 fn find_node_id(config: &Config) -> Result<String, UserError> {
     let node_id_path = Path::new(config.state_dir()).join("node_id");
 
-    // Check if node file exists
-    if node_id_path.exists() {
-        // If the node file exists, read the node_id within the file.
-        let mut file_node_id = fs::read_to_string(&node_id_path).map_err(|err| {
-            UserError::io_err_with_source("Unable to read node_id file", Box::new(err))
-        })?;
-        if file_node_id.ends_with('\n') {
-            file_node_id.pop();
-        }
-        match config.node_id() {
-            // If the config has a node_id, check if this matches the node_id read from the file.
-            Some(config_node_id) => {
-                if config_node_id != file_node_id {
-                    // If the node_id from the config object and the file do not match,
-                    // return an error.
-                    Err(UserError::InvalidArgument(format!(
-                        "node_id from file {} does not match node_id from config {}",
-                        file_node_id, config_node_id
-                    )))
-                } else {
-                    // If the node_id does match, then we return this node_id and continue.
-                    Ok(config_node_id.to_string())
+    #[cfg(feature = "node-file-block")]
+    {
+        if node_id_path.exists() {
+            let context = "node_id file is soft-deprecated, run splinter database migrate and splinter upgrade to import the value".to_string();
+            Err(UserError::DaemonError {
+                context,
+                source: None,
+            })
+        } else {
+            let database_uri = config
+                .database()
+                .parse()
+                .map_err(|_| UserError::InvalidArgument("db_connection".to_string()))?;
+            let store = create_store_factory(database_uri)?.get_node_id_store();
+            let db_node_id = store.get_node_id();
+            let config_node_id = config.node_id();
+            let save_new_node_id = |node_id| -> Result<(), UserError> {
+                store
+                    .set_node_id(node_id)
+                    .map_err(|err| UserError::from(InternalError::from_source(Box::new(err))))
```

Comment:
I think "node-file-block" needs to be added to the feature guard for `use splinter::error::InternalError;` on line 44, attempting to compile with just the `node-file-block` feature returns the error: "failed to resolve: use of undeclared type `InternalError`"